src/include/duckdb/planner/operator/logical_create_index.hpp

```diff
@@ -50,20 +52,19 @@ class LogicalCreateIndex : public LogicalOperator {
 	vector<string> names;
 	//! The types of ALL columns that can be returned by the table scan
 	vector<LogicalType> returned_types;
+	//! Types used for the CREATE INDEX scan
+	vector<LogicalType> scan_types;
```

Comment:
Are the `returned_types` still used? Only the `scan_types` are used no?

Also, shouldn't this information be in the `CreateIndexInfo`? Right now, the info needed to create the index is split between the `LogicalCreateIndex` and the `CreateIndexInfo`, and I think putting it all in the info makes the most sense.

If you have any thoughts on this, please let me know.